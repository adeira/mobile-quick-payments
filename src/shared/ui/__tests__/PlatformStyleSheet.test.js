// @flow

import { Platform } from 'react-native';

import StyleSheet from '../PlatformStyleSheet';

let originalPlatform;
beforeEach(() => (originalPlatform = Platform.OS));
afterEach(() => (Platform.OS = originalPlatform));

const createStyle = () =>
  StyleSheet.create({
    container: {
      flexDirection: 'row',
      ios: {
        padding: 10,
      },
      android: {
        padding: 20,
      },
    },
  });

it('works with iOS', () => {
  Platform.OS = 'ios';
  expect(createStyle()).toEqual({
    container: { flexDirection: 'row', padding: 10 },
  });
});

it('works with Android', () => {
  Platform.OS = 'android';
  expect(createStyle()).toEqual({
    container: { flexDirection: 'row', padding: 20 },
  });
});

it('works with unknown platform', () => {
  Platform.OS = 'unknown platform';
  expect(createStyle()).toEqual({
    container: { flexDirection: 'row' },
  });
});

it('exports absolute fill object', () => {
  expect(StyleSheet.absoluteFill).toMatchSnapshot();
  expect({
    ...StyleSheet.absoluteFill,
    position: 'relative',
  }).toMatchSnapshot();
});

it('exports hairline width', () => {
  expect(StyleSheet.hairlineWidth).toMatchSnapshot();
});
