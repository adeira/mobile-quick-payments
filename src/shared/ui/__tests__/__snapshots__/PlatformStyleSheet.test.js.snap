// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`exports absolute fill object 1`] = `
Object {
  "bottom": 0,
  "left": 0,
  "position": "absolute",
  "right": 0,
  "top": 0,
}
`;

exports[`exports absolute fill object 2`] = `
Object {
  "bottom": 0,
  "left": 0,
  "position": "relative",
  "right": 0,
  "top": 0,
}
`;

exports[`exports hairline width 1`] = `0.5`;
